### ISSUE TITLE
Search in not authorized folders throw an exception

### ISSUE THREAD

**Issue by farfilli** [rmcanany/SolidEdgeHousekeeper#155]
I have added my document folder as a source with subfolders search.
It happens that for some reason the image folder is not accessible.

The code should be smart enough to handle this situation

![image](https://github.com/user-attachments/assets/51c05392-7979-4cce-a4b5-bc706c5f2ba4)

**Comment by rmcanany** [comment]
Agreed.  I took a quick look at `FileIO.FileSystem.GetFiles` and didn't see an option to skip read-only directories.  I guess we could add a try/catch and report the error given.  At least it wouldn't crash that way.

**Comment by farfilli** [comment]
First search for all the subfolders then for each subfolder search files, in this way it's possible to handle single subfolder error.
Done in 495d90e6c0f60bbfa69c182b9b549560075a9257

**Comment by rmcanany** [comment]
Good idea.  Tell me if I'm wrong, but shouldn't `SearchAllSubDirectories` be `SearchTopLevelOnly`?
```
                        For Each tmpFolder In tmpFolders

                            Try

                                tmpFoundFiles.AddRange(FileIO.FileSystem.GetFiles(tmpFolder,
                                FileIO.SearchOption.SearchAllSubDirectories,
                                ActiveFileExtensionsList.ToArray))

                            Catch ex As Exception

                                Dim s As String = "An error occurred searching for files.  Please rectify the error and try again."
                                s = String.Format("{0}{1}{2}", s, vbCrLf, ex.ToString)
                                MsgBox(s, vbOKOnly)

                            End Try

                        Next
```

**Comment by farfilli** [comment]
You are right, I didn't pay attention to that part of the code. You do the fix?

**Comment by rmcanany** [comment]
I'll fix it.  Good point about the message box.  I'll move it outside the loop.

**Comment by rmcanany** [comment]
Fixed.  One question -- is a "TXT error message" the same as a MsgBox?

Incidentally, this was only finding the first level of subs.
`Dim tmpFolders As String() = Directory.GetDirectories(Source.Name)`

I changed it to this.
`Dim tmpFolders As String() = Directory.GetDirectories(Source.Name, "*", SearchOption.AllDirectories)`

**Comment by farfilli** [comment]
It's how you show errors at the end of files processing.

Good catch on the directory retrieval 👍🏻

**Comment by farfilli** [comment]
Unfortunately the SearchOption.AllDirectories works in the same way as files
![image](https://github.com/user-attachments/assets/005ef9d9-1dc0-4f80-9fc4-1cc0c0e9bb77)

**Comment by rmcanany** [comment]
Looks good.  

I see my error reporting is still not ideal.  It should wait until all files are processed before providing feedback.  

I'm going to leave it for now and add a note in the error-handling issue to take care of this when I do that.

**Comment by rmcanany** [comment]
Closing as complete.  Thank you!
